src/dune_pkg/rev_store.mli

```diff
@@ -21,15 +21,33 @@ module At_rev : sig
 end
 
 module Remote : sig
+  (** [uninit] represents an uninitialized remote. Use [update] or
+      [don't_update] to get a remote [t] *)
+  type uninit
+
+  (** handle representing a particular git repository *)
   type t
 
   val equal : t -> t -> bool
-  val update : t -> unit Fiber.t
+
+  (** [update remote] will fetch the most current revisions from the remote *)
+  val update : uninit -> t Fiber.t
+
+  (** [don't_update] signalizes that the remote should not be updated *)
+  val don't_update : uninit -> t
+
   val default_branch : t -> string
   val rev_of_name : t -> name:string -> At_rev.t option Fiber.t
   val rev_of_repository_id : t -> Repository_id.t -> At_rev.t option Fiber.t
 end
 
 val content_of_files : t -> File.t list -> string list Fiber.t
 val load_or_create : dir:Path.t -> t Fiber.t
-val add_repo : t -> source:string -> Remote.t Fiber.t
+
+(** [add_repo t ~source] idempotently registers a git repo to the rev store.
+    [source] is any URL that is supported by [git remote add].
+
+    Remember to run [Remote.update] if you want the remote repository to be
```

Comment:
The instructions in this comment are now enforced via `Remote.uninit`. Is it worth mentioning that here?

Reply:
That's only if you use the remote however.

Reply:
Good point, I wrote it before I enforced it. So I changed to wording a bit to be more clear and give people pointers what to do with a `Remote.uninit`, because it is still helpful to understand.